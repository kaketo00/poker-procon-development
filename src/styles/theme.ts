import { extendTheme } from '@chakra-ui/react';
import { Zen_Kaku_Gothic_New } from 'next/font/google';

const nextFont = Zen_Kaku_Gothic_New({
  weight: ['400', '700', '900'],
  subsets: ['latin'],
});

const theme = extendTheme({
  fonts: {
    heading: nextFont.style.fontFamily,
    body: nextFont.style.fontFamily,
  },
  colors: {
    brand100: {
      100: '#0DD203', // green 100%
      200: '#005CFF', // blue 100%
      300: '#B4171E', // red 100%
      600: '#777777', // dark-gray 100%
      700: '#D4D4D4', // gray 100%
      800: '#ECECEC', // light-gray 100%
    },
    brand80: {
      100: 'rgba(13, 210, 3, 0.8)', // green 80%
      200: 'rgba(0, 92, 255, 0.8)', // blue 80%
      300: 'rgba(180, 23, 30, 0.8)', // red 80%
      400: 'rgba(255, 255, 255, 0.8)', // white 80%
      500: 'rgba(0, 0, 0, 0.8)', // black 80%
      600: 'rgba(119, 119, 119, 0.8)', // gray 80%
    },
  },
  textStyles: {
    normal: {
      xs: {
        fontSize: '0.625rem',
        fontWeight: 'normal',
        lineHeight: '0.75rem',
      },
      sm: {
        fontSize: '0.75rem',
        fontWeight: 'normal',
        lineHeight: '1rem',
      },
      md: {
        fontSize: '1rem',
        fontWeight: 'normal',
        lineHeight: '1.25rem',
      },
      lg: {
        fontSize: '1.25rem',
        fontWeight: 'normal',
        lineHeight: '1.5rem',
      },
      xl: {
        fontSize: '1.5rem',
        fontWeight: 'normal',
        lineHeight: '1.75rem',
      },
    },
    bold: {
      xs: {
        fontSize: '0.625rem',
        fontWeight: 'bold',
        lineHeight: '0.75rem',
      },
      sm: {
        fontSize: '0.75rem',
        fontWeight: 'bold',
        lineHeight: '1rem',
      },
      md: {
        fontSize: '1rem',
        fontWeight: 'bold',
        lineHeight: '1.25rem',
      },
      lg: {
        fontSize: '1.25rem',
        fontWeight: 'bold',
        lineHeight: '1.5rem',
      },
      xl: {
        fontSize: '1.5rem',
        fontWeight: 'bold',
        lineHeight: '1.75rem',
      },
    },
    strong: {
      xs: {
        fontSize: '0.6875rem',
        fontWeight: 'black',
        lineHeight: '1rem',
      },
      sm: {
        fontSize: '1.25rem',
        fontWeight: 'black',
        lineHeight: '1.5rem',
      },
      md: {
        fontSize: '1.5rem',
        fontWeight: 'black',
        lineHeight: '2rem',
      },
      lg: {
        fontSize: '1.875rem',
        fontWeight: 'black',
        lineHeight: '2.375rem',
      },
      xl: {
        fontSize: '2.25rem',
        fontWeight: 'black',
        lineHeight: '2.75rem',
      },
    },
    heading: {
      h6: {
        fontSize: '1rem',
        fontWeight: 'black',
        lineHeight: '1.25rem',
      },
      h5: {
        fontSize: '1.25rem',
        fontWeight: 'black',
        lineHeight: '1.5rem',
      },
      h4: {
        fontSize: '1.5rem',
        fontWeight: 'black',
        lineHeight: '1.875rem',
      },
      h3: {
        fontSize: '1.75rem',
        fontWeight: 'black',
        lineHeight: '2.125rem',
      },
      h2: {
        fontSize: '2.25rem',
        fontWeight: 'black',
        lineHeight: '2.75rem',
      },
      h1: {
        fontSize: '1.875rem',
        fontWeight: 'black',
        lineHeight: '2.25rem',
      },
    },
    shadows: {
      light5: {
        100: '0px 0px 5px rgba(13, 210, 3, 0.8)',
        200: '0px 0px 5px rgba(0, 92, 255, 0.8)',
        300: '0px 0px 5px rgba(180, 23, 30, 0.8)',
        400: '0px 0px 5px rgba(255, 255, 255, 0.8)',
      },
      light10: {
        100: '0px 0px 10px rgba(13, 210, 3, 0.8)',
        200: '0px 0px 10px rgba(0, 92, 255, 0.8)',
        300: '0px 0px 10px rgba(180, 23, 30, 0.8)',
        400: '0px 0px 10px rgba(255, 255, 255, 0.8)',
      },
    },
  },
  breakpoints: {
    base: '0px', // 0 - 1279px game-template(S)
    sm: '80em', // 1280px - 1439px game-template(S)
    md: '90em', // 1440px - 1919px game-template(M)
    lg: '120em', // 1920px - 2559px game-template(M)
    xl: '160em', // 2560px - âˆž game-template(L)
    '2xl': '160em',
  },
});

export default theme;
