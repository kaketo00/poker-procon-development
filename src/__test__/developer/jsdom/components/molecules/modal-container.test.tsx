import { ChakraProvider } from '@chakra-ui/react';
import { render } from '@testing-library/react';
import type { ReactNode } from 'react';

import type { ModalContainerProps } from '@/components/molecules/modal-container';
import ModalContainer from '@/components/molecules/modal-container';
import theme from '@/styles/theme';

jest.mock('@chakra-ui/portal', () => ({
  Portal: jest.fn(({ children }: { children: ReactNode }) => children),
}));

const Component = (props: ModalContainerProps) => (
  <ChakraProvider theme={theme}>
    <ModalContainer {...props} />
  </ChakraProvider>
);

describe('ModalContainer', () => {
  test('スナップショット', () => {
    const { asFragment, getByText } = render(
      <Component
        title="モーダル"
        primaryLabel="Submit"
        secondaryLabel="Cancel"
        isOpen={true}
        onClose={jest.fn()}
      >
        モーダルコンテンツ
      </Component>
    );

    expect(getByText('モーダル')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
