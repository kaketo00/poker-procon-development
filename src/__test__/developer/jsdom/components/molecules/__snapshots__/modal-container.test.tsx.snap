// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalContainer スナップショット 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    class="chakra-modal__overlay css-hdd9l7"
    data-aria-hidden="true"
    style="opacity: 0;"
  />
  <div
    aria-hidden="true"
    data-aria-hidden="true"
    data-focus-guard="true"
    style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
    tabindex="0"
  />
  <div
    data-focus-lock-disabled="false"
  >
    <div
      class="chakra-modal__content-container css-1u2cvaz"
      tabindex="-1"
    >
      <section
        aria-describedby="chakra-modal--body-:r0:"
        aria-labelledby="chakra-modal--header-:r0:"
        aria-modal="true"
        class="chakra-modal__content css-pv22qu"
        id="chakra-modal-:r0:"
        role="dialog"
        style="opacity: 0; transform: scale(0.95) translateZ(0);"
        tabindex="-1"
      >
        <header
          class="chakra-modal__header css-9fgtzh"
          id="chakra-modal--header-:r0:"
        >
          モーダル
        </header>
        <button
          aria-label="Close"
          class="chakra-modal__close-btn css-1ik4h6n"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="chakra-icon css-onkibi"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
              fill="currentColor"
            />
          </svg>
        </button>
        <div
          class="chakra-modal__body css-qlig70"
          id="chakra-modal--body-:r0:"
        >
          モーダルコンテンツ
        </div>
        <footer
          class="chakra-modal__footer css-k0waxj"
        >
          <div
            class="css-1vakbk4"
          >
            <button
              class="chakra-button css-11494p6"
              type="button"
            >
              Cancel
            </button>
            <button
              class="chakra-button css-41aesz"
              type="button"
            >
              Submit
            </button>
          </div>
        </footer>
      </section>
    </div>
  </div>
  <div
    aria-hidden="true"
    data-aria-hidden="true"
    data-focus-guard="true"
    style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
    tabindex="0"
  />
  <div
    aria-label="Notifications-top"
    aria-live="polite"
    id="chakra-toast-manager-top"
    role="region"
    style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
  />
  <div
    aria-label="Notifications-top-left"
    aria-live="polite"
    id="chakra-toast-manager-top-left"
    role="region"
    style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
  />
  <div
    aria-label="Notifications-top-right"
    aria-live="polite"
    id="chakra-toast-manager-top-right"
    role="region"
    style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
  />
  <div
    aria-label="Notifications-bottom-left"
    aria-live="polite"
    id="chakra-toast-manager-bottom-left"
    role="region"
    style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
  />
  <div
    aria-label="Notifications-bottom"
    aria-live="polite"
    id="chakra-toast-manager-bottom"
    role="region"
    style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
  />
  <div
    aria-label="Notifications-bottom-right"
    aria-live="polite"
    id="chakra-toast-manager-bottom-right"
    role="region"
    style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
  />
  <span
    aria-hidden="true"
    data-aria-hidden="true"
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
